btf/marshal.go

```diff
@@ -464,6 +478,33 @@ func (e *encoder) deflateEnumValues(values []EnumValue) ([]btfEnum, error) {
 	return bes, nil
 }
 
+func (e *encoder) deflateEnum64(raw *rawType, enum *Enum) (err error) {
+	if e.ReplaceEnum64 {
+		// Replace an ENUM64 with an integer of the appropriate size. libbpf
+		// instead uses a union of (u)int64 which allows preserving the
+		// enum names (but not their values). Using the integer is simpler for
+		// us.
+		raw.SetKind(kindInt)
```

Comment:
Use `union` instead to match libbpf behaviour, otherwise we might create BTF that is incompatible with libbpf.